reproducer_metal_address_space.py

```
@@ -0,0 +1,111 @@
+
+    # Save for reference
+    with open("/tmp/metal_fixed_test.metal", "w") as f:
+        f.write(fixed)
+
+    # Verify with torch.mps.compile_shader
+    torch.mps.compile_shader = _original_compile
+    try:
+        result = _original_compile(fixed)
+        print("  ✅ Fixed source compiles successfully!")
+        print("  ✅ PR #2767 verification: PASS")
+    except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)